provisioner/inspec/provisioner.go

```diff
@@ -412,9 +418,33 @@ func (p *Provisioner) executeInspec(ui packersdk.Ui, comm packersdk.Communicator
 		return err
 	}
 	wg.Wait()
-	err = cmd.Wait()
-	if err != nil {
-		return fmt.Errorf("Non-zero exit status: %s", err)
+
+	if err := cmd.Wait(); err != nil {
+		if exiterr, ok := err.(*exec.ExitError); ok {
+			// The program has exited with an exit code != 0
+
+			// This works on both Unix and Windows. Although package
+			// syscall is generally platform dependent, WaitStatus is
+			// defined for both Unix and Windows and in both cases has
+			// an ExitStatus() method with the same signature.
+			if status, ok := exiterr.Sys().(syscall.WaitStatus); ok {
+				exitStatus := status.ExitStatus()
+				// Check exit code against allowed codes (likely just 0)
+				validExitCode := false
+				for _, v := range p.config.ValidExitCodes {
+					if exitStatus == v {
+						validExitCode = true
+					}
+				}
+				if !validExitCode {
+					return fmt.Errorf(
+						"Inspec exited with non-zero exit status: %d. Allowed exit codes are: %v",
```

Comment:
```suggestion
						"Inspec exited with unexpected exit status: %d. Allowed exit codes are: %v",
```

Reply:
Perhaps replace "Allowed exit codes" with "expected exit codes" as well, so it doesn't imply that only those codes are allowed.

Reply:
> Perhaps replace "Allowed exit codes" with "expected exit codes" as well, so it doesn't imply that only those codes are allowed.

Done, I just pushed the change.

Reply:
@azr I added your suggestion as well.